crates/anvil/tests/it/fork.rs

```diff
@@ -1203,3 +1203,92 @@ async fn test_fork_execution_reverted() {
     let err = resp.unwrap_err();
     assert!(err.to_string().contains("execution reverted"));
 }
+
+// <https://github.com/foundry-rs/foundry/issues/8227>
+#[tokio::test(flavor = "multi_thread")]
+async fn test_arbitrum_fork_transaction_hash() {
+    use std::str::FromStr;
+
+    // Fork to a block with a specific transaction
+    let hash =
+        TxHash::from_str("0x81ab9783151741b03b304f1874063da4f3263d0127c90ba500914d75e46b9c51")
+            .unwrap();
+    let (api, _) = spawn(
+        fork_config()
+            .with_blocktime(Some(Duration::from_secs(20)))
+            .with_fork_transaction_hash(Some(hash))
+            .with_eth_rpc_url(Some("https://arb1.arbitrum.io/rpc".to_string())),
+    )
+    .await;
+
+    // ...
+    let block_number = api.block_number().unwrap().to::<u64>();
+    assert_eq!(block_number, 224480116);
+
+    //let snapshot = api.evm_snapshot().await.unwrap();
+    api.mine_one().await;
+    let block_number = api.block_number().unwrap().to::<u64>();
+    assert_eq!(block_number, 224480117);
+
+    let block = api.block_by_number(BlockNumberOrTag::Number(224480116)).await.unwrap().unwrap();
+    assert_eq!(block.transactions.len(), 7);
+
+    // Validate the transactions preceding the target transaction exist
+    for expected_transactions in [
+        (
+            "0x0274d0c749cd9ef41103370ef46b20861aa43317aebcdc443f160d21a7895c90",
+            address!("00000000000000000000000000000000000a4b05"),
+        ),
+        (
+            "0xefe97eab0ef16ca12d9ebabe713a26670f7990a0d1653a28f218daf81a5bc9aa",
+            address!("e4edb277e41dc89ab076a1f049f4a3efa700bce8"),
+        ),
+        (
+            "0x53382f62959eb7907579cbdf5d3732a2eb0d2f9f597b1f0091ab7f3c31490d5c",
+            address!("f4cbc579caa87b4b6fc708497907cd053e5abb77"),
+        ),
+        (hash.to_string().as_str(), address!("31f9400360b89d96f63528db9d0446302ae64e71")),
+    ] {
+        let tx = api
+            .transaction_by_hash(TxHash::from_str(expected_transactions.0).unwrap())
+            .await
+            .unwrap()
+            .unwrap();
+        assert_eq!(tx.inner.to.unwrap(), expected_transactions.1);
+    }
+
+    // Validate the transactions following the target transaction don't exist
+    for unexpected_transactions in [
```

Comment:
@mattsse these tests fail atm because the api struct's eth rpc methods end up using the `provider` as a fallback if the transaction doesn't exist in the pool or local storage.

In order to test my changes, I need to update this behaviour to make the provider fallback optional. Unfortunately this will involve tampering with a few layers (api, backend, etc).

LMK if that sounds reasonable or if you know of an alternate approach